### PR TITLE
Notebook separation

### PR DESCRIPTION
**Pull request by SylvainCorlay**
- Exclude `static/notebook/js` from `jupyter-js-widgets` npm package.
- Move `save_state.js` and `manager.js` to `static/notebook/js`.

(The notebook-specific implementation of the widget manager should lie with notebook. Only manager-base should remain within jupyter-js-widgets.)

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
The only reason why we still have a webpack external to `notebook/js/outputarea` is the output widget.

**Comment by SylvainCorlay** [comment]
This is ready.

File: ipywidgets/static/notebook/js/extension.js [hunk 1]
```diff
@@ -25,10 +27,13 @@ define([
...
             // IPython.notebook handle.
-            var manager = new widgets.WidgetManager(kernel.comm_manager, IPython.notebook);
+            var manager = new mngr.WidgetManager(kernel.comm_manager, IPython.notebook);
+
+            // For backwards compatibility and interactive use.
+            IPython.WidgetManager = manager;
 
             // Store a handle to the manager so we know not to
-            // another for this kernel.  This also is a convinience
+            // another for this kernel.This also is a convenience
```
Comment (jdfreder) [inline ipywidgets/static/notebook/js/extension.js]: (Irrelevant comment) Why'd you remove the spaces between the two sentences?

File: ipywidgets/static/notebook/js/manager.js [hunk 2]
```diff
@@ -4,17 +4,16 @@
 define([
     "underscore",
     "backbone",
-    "./utils",
     "services/kernels/comm",
-    "./manager-base"
-], function (_, Backbone, utils, comm, managerBase) {
```
Comment (jdfreder) [inline ipywidgets/static/notebook/js/manager.js]: Why remove utils? (see below)

File: ipywidgets/static/notebook/js/manager.js [hunk 3]
```diff
@@ -79,7 +78,7 @@ define([
                         return WidgetManager._load_callback.call(that);
                     }).then(function(state) {
                         that.set_state(state);
-                    }).catch(utils.reject('Error loading widget manager state', true));
+                    }).catch(widgets.reject('Error loading widget manager state', true));
```
Comment (jdfreder) [inline ipywidgets/static/notebook/js/manager.js]: reject should be a function of utils, it's designed to chain promise rejections IIRC

**Comment by jdfreder** [comment]
Really one meaningful comment, otherwise, good to go

File: ipywidgets/static/notebook/js/manager.js (hunk 2 above)
Comment (SylvainCorlay) [inline ipywidgets/static/notebook/js/manager.js]: all the exports of utils are exported in `jupyter-js-widgets`. Here we are out of `jupyter-js-widgets`.

File: ipywidgets/static/notebook/js/extension.js (hunk 1 above)
Comment (SylvainCorlay) [inline ipywidgets/static/notebook/js/extension.js]: fixing this.

File: ipywidgets/static/notebook/js/manager.js (hunk 2 above)
Comment (jdfreder) [inline ipywidgets/static/notebook/js/manager.js]: o i c!  "widgets" not "widget"

**Comment by jdfreder** [comment]
Thanks!

File: ipywidgets/.npmignore [hunk 4]
```diff
@@ -6,6 +6,7 @@ docs/man/*.gz
 docs/source/api/generated
 docs/gh-pages
 static/components/
+static/notebook/js/
```
Comment (jdfreder) [inline ipywidgets/.npmignore]: Looking at this more, we shouldn't have done this.  When we do the small split, **no** javascript will live in ipywidgets.  The javascript, included notebook specific javascript, should be in the npm package available to install by ipywidgets or w/e.  I'll re-include this in #336

**Pull request merged**